packages/builds/src/node/error-parser.spec.ts

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { IErrorParser, IErrorMatcher, IErrorPattern } from "./error-parser";
+import * as fs from 'fs';
+import * as stream from 'stream';
+import * as chai from "chai";
+import "mocha";
+import * as chaiAsPromised from "chai-as-promised";
+import { BuildUtils } from "./build-spec-utils";
+import * as path from "path";
+
+import { testContainer } from "./inversify.spec-config";
+
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+// tsc error pattern
+// const tscErrorPattern: IErrorPattern = {
+//     "patternName": "tsc",
+//     "regexp": '(.*)\s\(([\d,]+)\):\s(.*)\((\d+)\)',
+//     "fileGroup": 1,
+//     "locationGroup": 2,
+//     "severityGroup": 99,
+//     "codeGroup": 4,
+//     "messageGroup": 3
+// };
+// const tscErrorMatcher: IErrorMatcher = {
+//     "name": "TypeScript compiler",
+//     "label": "TypeScript errors",
+//     "owner": "tsc",
+//     "fileLocation": "relative",
+//     "pattern": tscErrorPattern
+// };
+
+
+// gcc/g++ compiler
+const gccErrorPattern: IErrorPattern = {
+    "patternName": "gnu-c-cpp compiler",
+    "regexp": '(.*?):(\\d+:\\d+):?\\s*([Ee]rror|ERROR|[Ww]arning): (.*)',
+    "fileGroup": 1,
+    "locationGroup": 2,
+    "severityGroup": 3,
+    "codeGroup": 99,
+    "messageGroup": 4
+};
+const gccErrorMatcher: IErrorMatcher = {
+    "name": "gnu-c-cpp compiler",
+    "label": "gcc/g++ errors",
+    "owner": "gnu-c-cpp",
+    "fileLocation": "relative",
+    "pattern": gccErrorPattern
+};
+
+// gcc/g++ linker
+const gccLinkerErrorPattern: IErrorPattern = {
+    "patternName": "gnu-c-cpp linker",
+    "regexp": '(.*):(\\d+):\\s(.*\\s[tof]+\\s`.*\')',
+    "fileGroup": 1,
+    "locationGroup": 2,
+    "severityGroup": 99,
+    "codeGroup": 99,
+    "messageGroup": 3
+};
+const gccLinkerErrorMatcher: IErrorMatcher = {
+    "name": "gnu-c-cpp linker",
+    "label": "gcc/g++ errors",
+    "owner": "gnu-c-cpp",
+    "fileLocation": "absolute",
+    "pattern": gccLinkerErrorPattern
+};
+
+
+describe("error-parser", () => {
+    let parser: IErrorParser;
+    beforeEach(function () {
+        parser = testContainer.get<IErrorParser>(IErrorParser);
+    });
+
+    describe('parse simple gcc error log', () => {
+        it('verify compiler errors and warning are found', () => {
+            const logName: String = '/test-resources/' + 'error.txt';
+            const readStream: NodeJS.ReadableStream = fs.createReadStream(__dirname + logName);
```

Comment:
you don't have to specific local variable type like that.. logname = ' ' is OK same for stream